src/agent/debuglet.ts

```diff
@@ -113,32 +113,29 @@ const formatBreakpoints = function(
 };
 
 /**
- * CachedPromise class is designed to support debug agent in Google Cloud
- * Function (GCF). As GCF is a serverless environment and we wanted to make
- * sure debug agent could always capture the snapshots, we designed this class
- * so that GCF could wait until the listActiveBreakpoints event is completed
- * before throttling CPU to 0. CachedPromise has two member functions.
- * get() is to create or return an existing promise. resolve() will resolve
- * the promise at appropriate code location (in our case is after
- * listActiveBreakpoints is executed).
+ * CachedPromise stores a promise for a limited time. Its member function get()
+ * will initially create a promise, or create a promise when previous promise
+ * stales, then return that promise. resolve() will resolve the stored promise.
  */
 export class CachedPromise {
```

Comment:
Does `CachedPromise` need to be exported?

Reply:
It is exported for testing.